exec/java-exec/src/test/java/org/apache/drill/exec/store/parquet/TestParquetMetadataVersion.java

```diff
@@ -0,0 +1,130 @@
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.apache.drill.exec.store.parquet;
+
+import org.apache.drill.common.exceptions.DrillRuntimeException;
+import org.junit.Test;
+
+import static org.junit.Assert.assertEquals;
+import static org.junit.Assert.assertTrue;
+
+
+public class TestParquetMetadataVersion {
+
+  @Test
+  public void testCorrectOnlyMajorVersion() throws Exception {
+    MetadataVersion parsedMetadataVersion = new MetadataVersion(MetadataVersion.Constants.V1);
+    MetadataVersion expectedMetadataVersion = new MetadataVersion(1, 0);
+    assertEquals("Parquet metadata version is parsed incorrectly", expectedMetadataVersion, parsedMetadataVersion);
+  }
+
+  @Test
+  public void testCorrectMajorMinorVersions() throws Exception {
+    MetadataVersion parsedMetadataVersion = new MetadataVersion(MetadataVersion.Constants.V3_1);
+    MetadataVersion expectedMetadataVersion = new MetadataVersion(3, 1);
+    assertEquals("Parquet metadata version is parsed incorrectly", expectedMetadataVersion, parsedMetadataVersion);
+  }
+
+  @Test
+  public void testTwoDigitNumberMajorVersion() throws Exception {
+    String twoDigitNumberMajorVersion = "v10.2";
+    MetadataVersion parsedMetadataVersion = new MetadataVersion(twoDigitNumberMajorVersion);
+    MetadataVersion expectedMetadataVersion = new MetadataVersion(10, 2);
+    assertEquals("Parquet metadata version is parsed incorrectly", expectedMetadataVersion, parsedMetadataVersion);
+  }
+
+  @Test
+  public void testCorrectTwoDigitNumberMinorVersion() throws Exception {
+    String twoDigitNumberMinorVersion = "v3.10";
+    MetadataVersion parsedMetadataVersion = new MetadataVersion(twoDigitNumberMinorVersion);
+    MetadataVersion expectedMetadataVersion = new MetadataVersion(3, 10);
+    assertEquals("Parquet metadata version is parsed incorrectly", expectedMetadataVersion, parsedMetadataVersion);
+  }
+
+  @Test(expected = DrillRuntimeException.class)
+  public void testVersionWithoutFirstLetter() throws Exception {
+    String versionWithoutFirstLetter = "3.1";
+    try {
+      new MetadataVersion(versionWithoutFirstLetter);
+    } catch (DrillRuntimeException e) {
+      assertTrue("Not expected exception is obtained while parsing parquet metadata version",
+          e.getMessage().contains("Could not parse metadata"));
+      throw new DrillRuntimeException(e);
```

Comment:
1. `throw e`, no need to wrap it again in `DrillRuntimeException`
2. You can check that error message also contains corrupted version:
`contains(String.format("Could not parse metadata version %s", versionWithoutFirstLetter)));`

Please correct here and in tests below.